bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/automation/HueRuleConditionHandler.java

```diff
@@ -0,0 +1,240 @@
+            }
+            itemUID = entry.item.getUID();
+        } else if ("sensors".equals(validation[1]) && "state".equals(validation[3])) {
+            HueSensorEntry entry = ds.sensors.get(uid);
+            if (entry == null) {
+                throw new IllegalStateException("Sensor does not exist: " + uid);
+            }
+            itemUID = entry.item.getUID();
+        } else if ("config".equals(validation[1]) && "localtime".equals(validation[2])) {
+            // Item not used
+            itemUID = "";
+        } else {
+            throw new IllegalStateException("Can only handle groups and lights");
+        }
+
+        if (itemUID == null) {
+            throw new IllegalStateException("Can only handle groups and lights");
+        }
+
+        final String value = config.value;
+        switch (config.operator) {
+            case eq:
+                if (value == null) {
+                    throw new IllegalStateException("Equal operator requires a value!");
+                }
+                predicate = state -> {
+                    if (state instanceof Number) {
+                        return Integer.valueOf(value) == ((Number) state).intValue();
+                    } else if (state instanceof OnOffType) {
+                        return Boolean.valueOf(value) == (((OnOffType) state) == OnOffType.ON);
+                    } else if (state instanceof OpenClosedType) {
+                        return Boolean.valueOf(value) == (((OpenClosedType) state) == OpenClosedType.OPEN);
+                    }
+                    return state.toFullString().equals(value);
+                };
+                break;
+            case gt:
+                if (value == null) {
+                    throw new IllegalStateException("GreaterThan operator requires a value!");
+                } else {
+                    final Integer integer = Integer.valueOf(value);
+
+                    predicate = state -> {
+                        if (state instanceof Number) {
+                            return integer < ((Number) state).intValue();
+                        } else {
+                            return false;
+                        }
+                    };
+                }
+                break;
+            case lt:
+                if (value == null) {
+                    throw new IllegalStateException("LowerThan operator requires a value!");
+                } else {
+                    final Integer integer = Integer.valueOf(value);
+
+                    predicate = state -> {
+                        if (state instanceof Number) {
+                            return integer > ((Number) state).intValue();
+                        } else {
+                            return false;
+                        }
+                    };
+                }
+                break;
+            case in:
+            case not_in:
+                if (value == null) {
```

Comment:
The complete body is no very big for a switch, but feel free to pick that up another time